### PR TITLE
Make goal text NonEmpty

### PR DESCRIPTION
**Pull request by kostmo**
Here's how the 20 scenarios with missing goals were identified:

    find data/scenarios -name "*.yaml" -printf "%p\t" -exec yq '.objectives | map(has("goal")) | all' {} \; | grep false | cut -f1

### CODE REVIEW

File: src/Swarm/Game/ScenarioInfo.hs [hunk 1]
```diff
@@ -347,6 +347,7 @@ loadScenarioItem em path = do
   case isDir of
     True -> SICollection collectionName <$> loadScenarioDir em path
     False -> do
+      sendIO $ putStrLn $ unwords ["Loading path:", path]
```
Comment (byorgey) [inline src/Swarm/Game/ScenarioInfo.hs:350]: Seems like an unrelated debugging change?

**Review by byorgey — APPROVED** [review]
Seems reasonable to me.  But what's the motivation?

File: data/scenarios/Testing/201-require/533-reprogram-simple.yaml [hunk 2]
```diff
@@ -5,7 +5,9 @@ description: |
   the target robot doesn't have.
   https://github.com/swarm-game/swarm/pulls/533
 objectives:
-  - condition: |
+  - goal:
+      - Reprogram
```
Comment (byorgey) [inline data/scenarios/Testing/201-require/533-reprogram-simple.yaml:9]: Adding "fake" objective descriptions seems like an opportunity to better document the test suite.

Comment (xsebek) [reply data/scenarios/Testing/201-require/533-reprogram-simple.yaml:9]: Exactly, I sometimes inspect the tests from main menu and having a goal would be helpful.

**Review by xsebek — COMMENTED** [review]
I also wander what your motivation is, but requiring a goal description (even for tests) sounds reasonable to me. 🙂
